### ISSUE TITLE
ERROR in TypeError: compilation.emitAsset is not a function

### ISSUE THREAD

**Issue by Hannah-goodridge** [schneidmaster/sitemap-webpack-plugin#58]
This plugin seems to fail on the emitAsset function.

Using:
 "webpack": "4.28.0",
"sitemap-webpack-plugin": "^1.1.0",

usage:

```   
const SitemapPlugin = require('sitemap-webpack-plugin').default;
const paths = ['url/', 'url/url2'];

```
inside plugins:

```
new SitemapPlugin({ base: 'https://www.bob.com', paths })
```

Generates the following error:

`ERROR in TypeError: compilation.emitAsset is not a function`

**Comment by Hannah-goodridge** [comment]
Any help would be greatly received, thanks!

**Comment by schneidmaster** [comment]
Hi! I plugged your configuration into a test case but wasn't able to reproduce your issue. Could you share the full error stacktrace?

**Comment by Hannah-goodridge** [comment]
Heres the full error

```
ERROR in TypeError: compilation.emitAsset is not a function
    at /Users/ME/Sites/test-site/node_modules/sitemap-webpack-plugin/lib/index.js:378:41
    at Array.forEach (<anonymous>)
    at SitemapWebpackPlugin.<anonymous> (/Users/ME/Sites/test-site/node_modules/sitemap-webpack-plugin/lib/index.js:377:34)
    at step (/Users/ME/Sites/test-site/node_modules/sitemap-webpack-plugin/lib/index.js:86:23)
    at Object.next (/Users/ME/Sites/test-site/node_modules/sitemap-webpack-plugin/lib/index.js:67:53)
    at fulfilled (/Users/ME/Sites/test-site/node_modules/sitemap-webpack-plugin/lib/index.js:57:58)
    ```

**Comment by schneidmaster** [comment]
Ah weird! The only thing I can think would cause that is if you're somehow using webpack 3. If you open `node_modules/webpack/package.json` is it actually on version 4.x? If you can maybe share your `package.json` and your npm or yarn lockfile and I can see if anything looks awry?

**Comment by Hannah-goodridge** [comment]
@schneidmaster This has been resolved after upgrading my webpack to version 4.40.0 as compilation.emitAsset has been available since webpack 4.40.0.

**Comment by schneidmaster** [comment]
Ah, thanks for letting me know! I also published 1.1.1 (#60) to fix this for webpack 4 prior to 4.40.
